pkg/agent/route/route_linux.go

```diff
@@ -1385,24 +1380,23 @@ func (c *Client) AddClusterIPRoute(svcIP net.IP) error {
 		// collected. After this commit, there will be only one stale route after Antrea Agent started.
 		routes, err := c.listIPRoutesOnGW()
 		if err != nil {
-			return fmt.Errorf("error listing ip routes: %w", err)
+			klog.ErrorS(err, "error listing ip routes")
+			return
 		}
 		for i := 0; i < len(routes); i++ {
-			if routes[i].Gw.Equal(gw) && !routes[i].Dst.IP.Equal(svcIP) && routes[i].Dst.Contains(svcIP) {
+			if routes[i].Gw.Equal(gw) && !routes[i].Dst.IP.Equal(clusterIPCIDR.IP) && routes[i].Dst.Contains(clusterIPCIDR.IP) {
 				staleRoutes = append(staleRoutes, &routes[i])
 			}
 		}
 	}
 
 	// Remove stale routes.
 	for _, rt := range staleRoutes {
-		if err = c.netlink.RouteDel(rt); err != nil {
-			return fmt.Errorf("failed to uninstall stale ClusterIP route %s: %w", rt.String(), err)
+		if err := c.netlink.RouteDel(rt); err != nil {
+			klog.ErrorS(err, "failed to uninstall stale ClusterIP route", "stale route", rt)
 		}
 		klog.V(4).InfoS("Uninstalled stale ClusterIP route successfully", "stale route", rt)
```

Comment:
```suggestion
		klog.V(4).InfoS("Uninstalled stale ClusterIP route successfully", "route", rt)
```
This actually will be a key when it's appearing on log system. So better not to add empty space.